# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1376193860 14400
#      Sun Aug 11 00:04:20 2013 -0400
# Node ID f95bb8763f4ee813702b568bf0a3c679ea838e69
# Parent  d9ca0f44f1e7d5dddcb544f59a94f0262e1d5f07
Allow keyup/down messages to pass through to Qt

* libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp (QWinTerminalImpl::winEvent): Change WM_KEYUP/KEYDOWN to return false to pass through to Qt, Ctrl-C set to not send on to terminal.

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -1296,23 +1296,25 @@ void QWinTerminalImpl::wheelEvent (QWhee
 
 bool QWinTerminalImpl::winEvent (MSG* msg, long* result)
 {
   switch (msg->message)
     {
     case WM_KEYDOWN:
     case WM_KEYUP:
     //case WM_CHAR:
-      // Forward Win32 message to the console window
-      PostMessage (d->m_consoleWindow,
+      if ( GetKeyState ('C') == 0 || GetKeyState (VK_CONTROL) == 0)
+        {
+          PostMessage (d->m_consoleWindow,
                    msg->message,
                    msg->wParam,
                    msg->lParam);
-      result = 0;
-      return true;
+        }
+      // allow Qt to process messages as well, in case of shortcuts etc
+      return false;
     default:
       return false;
     }
 }
 
 //////////////////////////////////////////////////////////////////////////////
 
 void QWinTerminalImpl::scrollValueChanged (int value)
